submission-scripts/script-templates/magpie-hive

```diff
@@ -0,0 +1,81 @@
+#############################################################################
+# Hive configurations
+#############################################################################
+#
+# Should Hive be run
+#
+# Specify yes or no.  Defaults to no.
+#
+export HIVE_SETUP=yes
+
+# Version
+
+export HIVE_VERSION="HIVEVERSIONDEFAULT"
+
+export HIVE_PORT=HIVEDEFAULTPORT
+
+# Path to your Hive build/binaries
+#
+# This should be accessible on all nodes in your allocation. Typically
+# this is in an NFS mount.
+#
+# Ensure the build matches the Hadoop/HDFS version this will run against.
+#
+export HIVE_HOME="HIVEDIRPREFIX/base/apache-hive-${HIVE_VERSION}-bin"
+
+# Path to store data local to each cluster node, typically something
+# in /tmp.  This will store local conf files and log files for your
+# job.  If local scratch space is not available, consider using the
+# MAGPIE_NO_LOCAL_DIR_DIR option.  See README for more details.
+
+export HIVE_LOCAL_DIR="LOCALDIRPREFIX/hive"
+
+# Hive Clients will make a connection to a metastore server which in turn queries the datastore (PostgreSQL)
+# for metadata. This sets the location of the metastore
+export HIVE_METASTORE_DIR="LUSTREDIRPREFIX/hive/database/postgres/data"
+
+# Configurations for Postgres, used as a backend for HiveServer/Hive Metastore
+export POSTGRES_HOME="POSTGRESDIRPREFIX/base/pgsql"
+export PGPORT=POSTGRESDEFAULTPORT
+export PGDATA="LUSTREDIRPREFIX/hive/database/postgres/data"
+
+# DB name for this project
+export HIVE_DB_NAME=HIVEDEFAULTDBNAME
+
+###########################################################################
+# Hive Job/Run Configurations
+############################################################################
+
+# Set hive job for MAGPIE_JOB_TYPE = hive
+# Use 'testbench' to run benchmark testing for Hive via hive-testbench modules
+#
+export HIVE_MODE="testbench"
+
+# The following are configurations for running the hive-testbench software
+#
+# Path to the hive-testbench software
+export HIVE_TESTBENCH_DIR="TESTBENCHDIR/base/hive-testbench-hdp3"
+
+# Select the type of benchmark.  Options are 'tpcds' and 'tpch'
+# The TPC Benchmark H (TPC-H) is a decision support benchmark. It
+# consists of a suite of business oriented ad-hoc queries and 
+# concurrent data modifications.
+
+# The TPC Benchmark DS (TPC-DS) is a decision support benchmark that models several 
+# generally applicable aspects of a decision support system, 
+# including queries and data maintenance
+# More information can be found at the Transaction Processing Council website
+
+export HIVE_TESTBENCH_TYPE="tpch"
+
+# Select the Hive CLI version.  Options are 'hive' and 'beeline'.
```

Comment:
something I never thought to ask before b/c I don't know enough about hive, but why support this configuration?  Internally, does it matter to the user what Magpie chooses?  In some scripts you hardcode beeline.  Obviously if the user wishes to interact with Hive, they can script on their own.  But not sure why the user needs to select this?

Reply:
Originally I left this in because hive-testbench supported both, and currently Hive supports both.  At some point the hive-cli is going to be depreciated however I haven't seen anything in the documentation as to when?

Reply:
ok, no problem leaving it in.